pyleoclim/utils/tsutils.py

```diff
@@ -48,6 +48,11 @@
 
 SECONDS_PER_YEAR = 365.25 * 60  * 60 * 24
 
+MATCH_A  = ['y', 'yr', 'yrs', 'year', 'years']
+MATCH_KA = ['ky', 'kyr', 'kyrs', 'kiloyear', 'ka'] 
+MATCH_MA = ['ma', 'my','myr','myrs']
+MATCH_GA = ['ga', 'gy', 'gyr', 'gyrs']
```

Comment:
perhaps make these `frozenset`?

Reply:
makes sense. will do. 

I'm also not sure:
* how often these get used elsewhere in the codebase (would it be more appropriately placed elsewhere?)
* why `['yr BP', 'yrs BP', 'years BP']` aren't included as a separate list

Reply:
hi @kcpevey , we're on it. 
The function tackles a few cases at once, and I am grateful for your review as there might be cases that fall throiugh the cracks. 
> how often these get used elsewhere in the codebase (would it be more appropriately placed elsewhere?)

it's mostly on series creation (`__init__`) and in `convert_time_unit()` (a Series API function), as well as a few other places. I don't think it matters much where they are placed as long as they are accessible. 

> why ['yr BP', 'yrs BP', 'years BP'] aren't included as a separate list

Good question. The issue is that the way most paleo folks report their units is context-dependent and conveys several pieces of information at once. BP = before present 0(which can mean either 1950, 2000, or the date of collection), so it's ambiguous, but always means retrograde time and (~90% of the time) `datum = 1950`.  "years" is a statement about the `exponent` parameter (0 in this case).  However, to complicate matters, the way the variable is called (usually Time or Age) also conveys some information: Age implies retrograde time; Time is more ambiguous.   

Currently the code tries to catch most commonly used idioms, but it will never be 100%, which is why it's necessary to set some sensible defaults, even if the user has to change them after the fact if they didn't get what they wanted.  Feel free to suggest other approaches if you can see a better way to do this.

Reply:
Thanks, I understand know. The `MATCH_*` statements are specifically about the exponent, but `yr BP` is a mix - BP is about the *datum* while years is about the *exponent*. 

>I don't think it matters much where they are placed as long as they are accessible.

Right. And "accessible" also includes "doesn't cause and import loop" - just something to keep in mind in case it comes up in the future.